### ISSUE TITLE
Clicking Next or last in student management section loads forever

### ISSUE THREAD

**Issue by toyinogun** [gocodebox/lifterlms#1578]
### Reproduction Steps

Go to the staging site on the ticket and go to any course.
Scroll down to the "Student Management" section
Click on Next or Last.

### Expected Behavior

Should move to the next/last page for students

### Actual Behavior

Just keeps rolling. 
Disabled plugins but keeps doing the same. However, I saw some errors in the error log that may be helpful.

### Error Messages / Logs

+ Include any relevant error messages or log files
```
[29-Mar-2021 07:16:03 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:16:18 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:16:34 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:16:55 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:17:02 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:17:03 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:17:12 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:17:21 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139
[29-Mar-2021 07:17:29 UTC] PHP Warning:  The symlink at <code>/home/ocgd3/public_html/classroom/staging/wp-content/db.php</code> is no longer pointing to the correct location. Please remove the symlink, then deactivate and reactivate Query Monitor. in /home/ocgd3/public_html/classroom/staging/wp-content/plugins/query-monitor/classes/QueryMonitor.php on line 139

```

### System and Environment Information

<details>
<summary>System Report</summary>

```
Wordpress
-------------------------------------------

Home Url: [removed]
Site Url: [removed]
Login Url: [removed]/wp-login.php
Version: 5.7
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_GB
Multisite: No
Page For Posts: Not Set
Page On Front: Course Catalogue (#1634) [[removed]/]
Permalink Structure: /%postname%/
Show On Front: page
Wp Cron: Yes


Settings
-------------------------------------------

Version: 4.20.0
Db Version: 
Course Catalog: About (#27) [[removed]/about/]
Membership Catalog: Membership Catalog (#3965) [[removed]/memberships/]
Student Dashboard: Dashboard (#115) [[removed]/dashboard/]
Checkout Page: Purchase (#114) [[removed]/purchase/]
Course Catalog Per Page: 0
Course Catalog Sorting: menu_order,ASC
Membership Catalog Per Page: 9
Membership Catalog Sorting: menu_order
Site Membership: Not Set
Courses Endpoint: my-courses
Edit Endpoint: edit-account
Lost Password Endpoint: lost-password
Vouchers Endpoint: 
Autogenerate Username: yes
Password Strength Meter: yes
Minimum Password Strength: medium
Terms Required: yes
Terms Page: Terms and Conditions (#2282) [[removed]/terms-and-conditions/]
Checkout Names: required
Checkout Address: optional
Checkout Phone: hidden
Checkout Email Confirmation: no
Open Registration: no
Registration Names: required
Registration Address: optional
Registration Phone: hidden
Registration Voucher: optional
Registration Email Confirmation: no
Account Names: required
Account Address: optional
Account Phone: optional
Account Email Confirmation: no
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: no
Country: GB
Currency: GBP
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: no
Recurring Payments: no
Email From Address: [removed]
Email From Name: [removed]
Email Footer Text: 
Email Header Image: 
Cert Bg Width: 800
Cert Bg Height: 616
Cert Legacy Compat: no


Constants
-------------------------------------------

LLMS_REMOVE_ALL_DATA: undefined
LLMS_REST_DISABLE: undefined
LLMS_SITE_FEATURE_RECURRING_PAYMENTS: undefined
LLMS_SITE_IS_CLONE: undefined


Gateways
-------------------------------------------

Manual: Disabled
Manual Logging: 
Manual Order: 1


Server
-------------------------------------------

Mysql Version: 5.5.5
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 1000
Php Max Upload Size: 8 MB
Php Memory Limit: 256M
Php Post Max Size: 8M
Php Soap: No
Php Suhosin: No
Php Time Limt: 30
Php Version: 7.4.16
Software: Apache
Wp Memory Limit: 40M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36


Theme
-------------------------------------------

Name: YOOtheme
Version: 2.4.5
Themeuri: https://yootheme.com
Authoruri: https://yootheme.com
Template: 
Child Theme: No
Llms Support: No


Plugins
-------------------------------------------

LifterLMS: 4.20.0


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No


Template Overrides
-------------------------------------------




```

</details>

This issue has be recreated:
+ [ ] Locally
+ [ ] On a staging site
+ [ ] On a production website
+ [ ] With only LifterLMS and a default theme
HS-160898

**Comment by thomasplevy** [comment]
@toyinogunseinde these errors point to another plugin (Query Monitor) and seem to indicate that there'd be more widespread problems than the one identified here.

Have we been able to recreate this anywhere?

**Comment by toyinogun** [comment]
@thomasplevy No, I'm unable to recreate this. I also see some errors in the console:
![image](https://user-images.githubusercontent.com/62220599/112870649-55087f80-90b6-11eb-9f17-29cbd45ec320.png)

**Comment by thomasplevy** [comment]
Thanks for the additional logs but this, unfortunately, doesn't really point to anything

I've logged into the customers stage and found that this appears to be a WordFence related error

The right side shows the response of the AJAX request, which has been blocked by WordFence.

Sometimes WordFence will incorrectly block requests. It does a lot of things by looking at common attacks and blocking things that *may look* like an attack. AJAX requests are a common place for malicious requests and I guess this one, on this particular site, looks like a potential attack. 

WordFence has the ability to safelist requests like this. I'd point the customer to these docs: https://www.wordfence.com/help/firewall/options/ -- the section titled "Monitor Background Requests for False Positives" is where to start.

Although... WordFence is *not enabled* (or even installed and deactivated). My guess is that maybe the live site has this enabled and it's protecting the stage installed in a subdirectory.

Please relay to the customer (let me know if you have questions or need help).

This seems like the issue but I'm not 100% positive since WordFence isn't actively installed. I'm not positive this is the exact issue that's happening on the production site but it's certainly what's causing this on the stage.

**Comment by toyinogun** [comment]
@thomasplevy The customer is still facing this issue and I'm blank as to what to do. Please help with this. Thanks.

**Comment by thomasplevy** [comment]
@toyinogunseinde I've just looked at this again and the issue reported here appears to be resolved to me. I went to the coures in question and clicked "Next" and it loaded as expected. No more WF error in the console. The customer said they disabled WordFence but they're still having this issue but I'm not having this issue as described anymore... I'll need more details to debug further as it looks fine from my end.

**Comment by thomasplevy** [comment]
@toyinogunseinde I've close this issue but it seems the related customer ticket is still open, could you please follow up with them ASAP!

**Comment by nrherron92** [comment]
@thomasplevy they've opened a new ticket on "the same" though I'd call it similar issue. They've swapped out WordFence for iThemes and there's a bit of a different issue. Student reporting and courses take a long time to load but eventually do load. Once I've loaded them I can move between them without the same speed issue. 

There is a potential plugin conflict I'm trying to get staging for and I've asked them about their host. Who hosts this said it was Node4 Limited 

I don't know if there's anything you need to look into again just yet but can I leave this reopened on standby for now?

**Comment by thomasplevy** [comment]
@nrherron92 

So they're not having the problem described in this bug report anymore but they're having a speed issue that you want me to look into?

What's the plugin conflict?

**Comment by nrherron92** [comment]
@thomasplevy I'm opening a github issue for the Javascript thing. 

I think we can just close this. I don't see any speed issues on the staging site and I'm not really experiencing them on the live one aside from that first load. I think this may be something they just need to talk to their host about.

**Comment by nrherron92** [comment]
@thomasplevy I am pretty sure this is just an issue with their server not being able to handle a heavier plugin. I do not see any javascript or php errors except for a fatal error for exhaustion from back in April. 

I do see there is a definite speed issue. While I don't see it on reporting he has mentioned he also cannot open courses. Again, with this, there is a long load time but the course will open for me and after the first load it loads faster when I come back to it.

I am at a point where I'm ready to tell him to contact his host (he really, really, doesn't want to) but can you take a look and see if I'm just blindly missing something? (Or just tell me you're busy and to stop being ridiculous and send the email to contact his host :P )

**Comment by thomasplevy** [comment]
@nrherron92 according to the system report the PHP memory limit is 256 and the WP memory limit is 40m

This is pretty standard and WP does boost the memory limit automatically on the admin panel but there's a possibility that there is a memory issue that could be resolved by increasing the resources allotted to the server.

Even if LifterLMS queries are the bottleneck, we're quite aware that LifterLMS uses a lot of resources. Loading student data is expensive and we do attempt to cache this reporting data wherever possible but in the event of the cache not being "primed" these screens are expected to take longer (student data is reported on the course page in the student management metabox).

Once the first slow page load happens though the cache is primed and you shouldn't see future issues. Background processing will take care of keeping it primed.

If the cache is somehow disabled or they're still having excessively slow page loads the next step is to contact the host.

The solution is likely more of a "You maybe want to consider a WP specifc host" because my guess is the host will tell them WordPress is slow or something. But who knows...

The website for node4 does mention managed hosting so they should have support from them in some capacity. I've never heard of node4 before though so I don't know.

I don't know what else to really debug here and call a bug. It sounds to me like an underperforming or outgrown server.

**Comment by nrherron92** [comment]
Thank you! That's what I thought!
